types/index.d.ts

```diff
@@ -131,13 +132,14 @@ declare module 'mongoose' {
    */
   export function isObjectIdOrHexString(v: any): boolean;
 
-  export function model<T>(name: string, schema?: Schema<T, any, any> | Schema<T & Document, any, any>, collection?: string, options?: CompileModelOptions): Model<T>;
```

Comment:
Why drop this declaration? This would be backwards breaking

Reply:
It's not backwards breaking by far, but it affects TS performance actually, so I've restored it.